radicle-cli/src/git.rs

```diff
@@ -175,24 +217,33 @@ pub fn is_signing_configured(repo: &Path) -> Result<bool, anyhow::Error> {
 }
 
 /// Return the list of radicle remotes for the given repository.
-pub fn remotes(repo: &git2::Repository) -> anyhow::Result<Vec<(String, NodeId)>> {
-    let mut remotes = Vec::new();
-
-    for name in repo.remotes().iter().flatten().flatten() {
-        let remote = repo.find_remote(name)?;
-        for refspec in remote.refspecs() {
-            if refspec.direction() != git2::Direction::Fetch {
-                continue;
-            }
-            if let Some((peer, _)) = refspec.src().and_then(self::parse_remote) {
-                remotes.push((name.to_owned(), peer));
-            }
-        }
-    }
-
+pub fn rad_remotes(repo: &git2::Repository) -> anyhow::Result<Vec<Remote>> {
+    let remotes: Vec<_> = repo
+        .remotes()?
+        .iter()
+        .filter_map(|name| {
+            let remote = repo.find_remote(name?).ok()?;
+            match Remote::try_from(remote) {
+                // ignore URLs that are not rad://
+                Err(RemoteError::ParseUrl(UrlError::UnsupportedScheme)) => Ok(None),
+                // raise an error if there's an invalid URL
+                Err(err) => Err(err), // I think this actually won't work because we're in `filter_map`
+                Ok(remote) => Ok(Some(remote)),
+            }.ok()?
```

Comment:
Let's go with what you originally had and add a FIXME
```suggestion
            // FIXME(finto): It would be better to surface malformed errors using
            // Iterator<Item = Result<Remote, RemoteError>>
            Remote::try_from(remote).ok()
```